torch/csrc/jit/ir/ir.cpp

```diff
@@ -1917,6 +1917,22 @@ std::vector<Value*> inlineCallTo(
   // are missing nodes without outputs (e.g. prim::Print).
   std::unordered_set<Node*> updated_nodes;
   for (const auto& kv : value_map) {
+    /* Skip the old value if it is the graph input.
+     * The reason is that, value_map contains values not all for the nodes of
+     * the graph but primary inputs as well, and it will create duplicates when
+     * the first inlined graph is input to the next one. To avoid this issue,
+     * skip the old value when it is one of the
+     * callee->optimized_graph()->inputs(). Use optimized_graph() here because
+     * it was used to generate value_map.
+     */
+    auto is_graph_input = std::find(
+        callee->optimized_graph()->inputs().begin(),
```

Comment:
I just realized, we should condition this on `inline_optimized_graph` since optimized_graph is only used when that is true.

Reply:
Good catch. In the last commit, check `optmized_graph` when `inline_optimized_graph`, otherwise check `graph`.